### ISSUE TITLE
Schema type definition is not executable

### ISSUE THREAD

**Issue by D19cannon** [apollographql/react-apollo#3772]
Hey guys,

I am completely new to Apollo. Currently, I am setting up a Headless WP, Gatsby, Graphql, Apollo project (e-commerce). However, for some reason I ran into an interesting problem while setting up the local state with Apollo.

I created the local state file as follow:

import { graphql } from 'gatsby';

export const typeDefs = graphql`
type CartItem {
id: ID!
sku: String!
title: String!
price: Int!
image: String
quantity: Int!
}

extend type Query {
isLoggedIn: Boolean!
cartItems: [CartItem]!
}

extend type Mutation {
addOrRemoveFromCart(id: ID!): [CartItem]
}
`;

export const resolvers = {};

// and use them in my apolloClient as follow (partial code):
import ApolloClient from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import { CachePersistor } from 'apollo-cache-persist';
import fetch from 'isomorphic-fetch';
import { resolvers, typeDefs } from './localState';

const client = new ApolloClient({
link: httpLink,
cache,
typeDefs,
resolvers,
});

You probably get the idea of what I am trying to setup in here. But for some reason it is returning the following error:

"error The CartItem definition is not executable graphql/template-strings"

I am not querying it anywhere or anything, just setting up my local state step by step. Also, the graphql/template-strings seems a bit confusing for me. I am not querying anything to that end point. My end-point is: http://wp.test/graphql/ which is the graphql schema in WP.

Let me know what your thoughts are on how to solve this.
Thank you in advance :)

Quick update
What I am trying to accomplish is basically this, but then on the font and which the cache etc.
https://codesandbox.io/s/apollo-server

**Comment by dylanwulf** [comment]
Are you using eslint-plugin-graphql? If so, you just need to manually disable the `ExecutableDefinitions` rule. Not sure why it ships with that rule enabled by default, but I don't think there's anything wrong with your code. This is what my `.eslintrc.js` file looks like:
```js
const { specifiedRules } = require('graphql');
const schemaJson = require('./test/schema.json');

const allGraphQLValidatorNames = specifiedRules.map(rule => rule.name);

module.exports = {
    rules: {
        // ................................
        // other irrelevant rules truncated for brevity
        // ................................
        'graphql/template-strings': [
            'error',
            {
                env: 'apollo',
                schemaJson,
                validators: without(
                    allGraphQLValidatorNames,
                    'KnownFragmentNames',
                    'NoUnusedFragments',
                    'NoUnusedVariables',
                    'ExecutableDefinitions',
                ),
            },
        ],
        'graphql/required-fields': [
            'error',
            {
                env: 'apollo',
                schemaJson,
                requiredFields: ['id'],
            },
        ],
        'graphql/named-operations': ['error', { schemaJson }],
    },
};
```

More information can be found on the [eslint-plugin-graphql github page](https://github.com/apollographql/eslint-plugin-graphql)

P.S. If you surround your code with triple backticks (` ``` `) before pasting, then the indentation and stuff gets preserved

**Comment by D19cannon** [comment]
Hey @dylanwulf, 

Thanks for your response. I've looked at your suggestion, but I a not using any linting anywhere (also checked my globals incase just incase). The  eslint-plugin-graphql github page is not a plugin that I am currently using. 

Also, I am not using any schema.json file. The schema is coming from WP with Gatsby and is thrown away afterwards once all the queries are made and pages/posts etc are created.

I will create an ESlint and see if it works, but I am having my doubts.

I will try in the linting in the meantime if you have any other suggestions, please throw them at me.

**Comment by dylanwulf** [comment]
That's very strange, I'm pretty confident that "graphql/template-strings" errors can only come from eslint-plugin-graphql. Either way though, I don't think there's anything wrong with your code.

**Comment by D19cannon** [comment]
@dylanwulf  I got another funky error with it.

import React, { useState, useEffect } from 'react'
import { useQuery, useMutation, useApolloClient } from '@apollo/react-hooks'

```
  48:18  error    React Hook "useApolloClient" is called in function "checkoutPage" which is neither a React function component or a custom React Hook function  react-hooks/rules-of-hooks
  61:20  error    React Hook "useQuery" is called in function "checkoutPage" which is neither a React function component or a custom React Hook function         react-hooks/rules-of-hooks
```

**Comment by D19cannon** [comment]
Quick update:

The following error has occured when trying to add @dylanwulf eslint configurations to the Gatsby Webpack config:

```
// webpack config

// webpack config
exports.onCreateWebpackConfig = ({
  stage,
  rules,
  loaders,
  plugins,
  actions,
}) => {
  actions.setWebpackConfig({
    module: {
        rules: [
          {
            // ................................
            // other irrelevant rules truncated for brevity
            // ................................
            'graphql/template-strings': [
                'error',
                {
                    env: 'apollo',
                    validators: without(
                        allGraphQLValidatorNames,
                        'KnownFragmentNames',
                        'NoUnusedFragments',
                        'NoUnusedVariables',
                        'ExecutableDefinitions',
                    ),
                },
            ],
            'graphql/required-fields': [
                'error',
                {
                    env: 'apollo',
                    requiredFields: ['id'],
                },
            ],
            'graphql/named-operations': ['error', { localstate }],
        },
      ],
    },
    plugins: [
      plugins.define({
        __DEVELOPMENT__: stage === `develop` || stage === `develop-html`,
      }),
    ],
  })
}
```

```
// error 
Error in "/Users/zer0cream/projects/gatsy-wp-template-ecommerce/gatsby-template/gatsby-node.js": Cannot use import statement outside a module



  Error: /Users/zer0cream/projects/gatsy-wp-template-ecommerce/gatsby-template/sr  c/utils/localState.js:1
  (function (exports, require, module, __filename, __dirname) { import { graphql   } from 'gatsby';
                                                                ^^^^^^
  SyntaxError: Cannot use import statement outside a module
```

**Comment by dylanwulf** [comment]
No, eslint config doesn't go inside of the webpack config. I don't think those other errors are related to the "not executable" error. I'm still pretty sure that the "not executable" error is coming from an eslint config somewhere which somehow uses eslint-plugin-graphql. Those other errors are due to some other setup issue.

**Comment by D19cannon** [comment]
Alright I understand @dylanwulf. I found some documentation about in gatsby documentation
https://www.gatsbyjs.org/packages/gatsby-plugin-lint-queries/

But what I don't understand is that I do not have a schema.json file. It is basically generated on build from Wordpress and thrown away later. However, it is only complaining about that one localstate.js file. How am I suppose to implement that?

**Comment by D19cannon** [comment]
WHOOP WHOOP!! IT WORKED!! 
Solution is basically just that link: https://www.gatsbyjs.org/packages/gatsby-plugin-lint-queries/

SWEET!

Thank you @dylanwulf for all your help and suggestions! 👍

**Comment by dylanwulf** [comment]
Sorry, I don't know. I've never used gatsby or wordpress before.
